addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -476,3 +479,38 @@ def _get_move_applicability(self, move):
             'post': self._l10n_sa_post_zatca_edi,
             'edi_content': self._l10n_sa_get_invoice_content_edi,
         }
+
+    def _prepare_invoice_report(self, pdf_writer, edi_document):
+        """
+        Prepare invoice report to be printed.
+        :param pdf_writer: The pdf writer with the invoice pdf content loaded.
+        :param edi_document: The edi document to be added to the pdf file.
+        """
+        super()._prepare_invoice_report(pdf_writer, edi_document)
```

Comment:
Nitpick but I like when we add a `self.ensure_one()` at the beginning of that kind of function, feel free to ignore.